src/nrniv/nrncore_write/callbacks/nrncore_callbacks.cpp

```diff
@@ -678,10 +678,10 @@ int nrnthread_dat2_vecplay_inst(int tid,
 
     PlayRecList* fp = net_cvode_instance->fixed_play_;
     if (fp->item(i)->type() == VecPlayContinuousType) {
-        VecPlayContinuous* vp = (VecPlayContinuous*) fp->item(i);
-        if (vp->discon_indices_ == NULL) {
+        auto* const vp = static_cast<VecPlayContinuous*>(fp->item(i));
+        if (!vp->discon_indices_) {
             if (vp->ith_ == nt.id) {
-                double* pd = vp->pd_;
+                auto* pd = static_cast<double*>(vp->pd_);
```

Comment:
There are a few lines like this that I think are no-ops (because the argument of the cast is currently a double* but will soon be a data_handle<double>). Probably it's not worth removing them here.

Reply:
> I think we have picked out a bit more than just the Datum changes

Indeed, I tried to get as much as possible for diff-reducing purposes